core/config.go

```diff
@@ -0,0 +1,345 @@
+package core
+
+import (
+	"errors"
+	"log"
+	"sync"
+	"time"
+
+	"github.com/Tapjoy/dynamiq/app/compressor"
+	"github.com/Tapjoy/dynamiq/app/stats"
+	"github.com/hashicorp/memberlist"
+)
+
+// StatsConfig represents config info for sending data to any statsd like system
+// and the client itself
+type StatsConfig struct {
+	Address       string
+	FlushInterval int
+	Prefix        string
+	Client        *stats.Client
+}
+
+// HTTPConfig represents config info for the HTTP server
+type HTTPConfig struct {
+	APIVersion string
+	Port       uint16
+}
+
+// DiscoveryConfig represents config info for how Dynamiq nodes discovery eachother via Memberlist
+type DiscoveryConfig struct {
+	Port       int
+	Memberlist *memberlist.Memberlist
+}
+
+// RiakConfig represents config info and the connection pool for Riak
+type RiakConfig struct {
+	Addresses          []string
+	Port               uint16
+	Service            *RiakService
+	ConfigSyncInterval time.Duration
+}
+
+// Config is the parent struct of all the individual configuration sections
+type Config struct {
+	Riak       *RiakConfig
+	Discovery  *DiscoveryConfig
+	HTTP       *HTTPConfig
+	Stats      *StatsConfig
+	Compressor *compressor.Compressor
+	Queues     *Queues
+	Topics     *Topics
+	// Channels / Timer for syncing the config
+	syncScheduler *time.Ticker
+	syncKiller    chan bool
+}
+
+var (
+	//ErrUnknownTopic is
+	ErrUnknownTopic = errors.New("There is no known topic by that name")
+
+	//ErrUnknownTopic is
+	ErrUnknownQueue = errors.New("There is no known queue by that name")
+)
+
+// GetConfig Parses and returns a config object
+func GetConfig() (*Config, error) {
+	// TODO settle on an actual config package
```

Comment:
At this point I've come to the general conclusion that there is no universal way to do config in golang, and alot of times it comes down to either rolling you're own or pulling a package that works mostly as you want.